### DISCUSSION TITLE
Global pytest timeout

### DISCUSSION THREAD

**Discussion by jondequinor** [equinor/ert#3344] [Ideas]
1. We should acknowledge that ERT will be flaky for a long time. Its design implies a lot of deeply buried synchronization issues that are exacerbated by using files writes/reads as means for communication.
1. Furthermore, our test suite runs for a very long time, and we want to cut that down.

Given these two assumptions, should we have a global [pytest timeout](https://pypi.org/project/pytest-timeout/)?

```ini
[pytest]
timeout = 60
```

Integration tests currently in conflict with this could be exempted from this policy by setting the timeout to 0 (or some higher number):

```python
@pytest.mark.timeout(0)
def test_integration():
    pass
```

**Comment by sondreso** [comment]
I think this is a good idea! It gives us better insight into which specific tests that are running to long, compared to the global timeout we have on Github Actions today. It will also give us a good opportunity to actually find and mark all test that are integration tests.

**Comment by frode-aarstad** [comment]
I think is a good idea too. By explicitly stating the timeout for a test the programmer can give an indication if a test is supposed to be fast or slow.

**Comment by pinkwah** [comment]
There seems to be unanimous agreement. Should this discussion be converted into an issue? (It's one of the options on the right)
